storage/snapshot/snapshot.rs

```diff
@@ -195,8 +198,9 @@ where
 }
 
 pub struct ReadSnapshot<D> {
-    storage: Arc<MVCCStorage<D>>,
     open_sequence_number: SequenceNumber,
+    iterator_pool: IteratorPool, // Must be declared & dropped before storage
+    storage: Arc<MVCCStorage<D>>,
```

Comment:
Alternately, the IteratorPool must hold a reference. This make the signature dependent on `<D>` which is ugly.